infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeploymentsTest.java

```diff
@@ -672,4 +685,100 @@ public void shouldHandleEventWithEmptyLastTimestampAndFirstTimestamp() throws Ex
     PodEvent podEvent = captor.getValue();
     assertNotNull(podEvent.getLastTimestamp());
   }
+
+  @Test
+  public void deploymentShouldHaveImagePullSecretsOfSAAndSelf() throws InfrastructureException {
+    doReturn(serverMock.getClient()).when(clientFactory).create(anyString());
+
+    final String serviceAccountName = "workspace-sa";
+    LocalObjectReference pullSecretOfSA =
+        new LocalObjectReferenceBuilder().withName("pullsecret-sa").build();
+    LocalObjectReference pullSecretOfPod =
+        new LocalObjectReferenceBuilder().withName("pullsecret-pod").build();
+    List<LocalObjectReference> imagePullSecrets = new ArrayList<>();
+    imagePullSecrets.add(pullSecretOfPod);
+
+    ServiceAccount serviceAccount =
+        new ServiceAccountBuilder()
+            .withMetadata(
+                new ObjectMetaBuilder()
+                    .withName(serviceAccountName)
+                    .withNamespace(kubernetesDeployments.namespace)
+                    .build())
+            .withImagePullSecrets(pullSecretOfSA)
+            .build();
+    serverMock
+        .getClient()
+        .serviceAccounts()
+        .inNamespace(kubernetesDeployments.namespace)
+        .create(serviceAccount);
+
+    ObjectMeta objectMeta =
+        new ObjectMetaBuilder()
+            .withName("test-pod")
+            .withNamespace(kubernetesDeployments.namespace)
+            .build();
+
+    Deployment deployment =
+        new DeploymentBuilder()
+            .withMetadata(objectMeta)
+            .withSpec(
+                new DeploymentSpecBuilder()
+                    .withNewTemplate()
+                    .withMetadata(objectMeta)
+                    .withNewSpec()
+                    .withImagePullSecrets(imagePullSecrets)
+                    .withServiceAccountName(serviceAccountName)
+                    .endSpec()
+                    .endTemplate()
+                    .build())
+            .build();
+
+    kubernetesDeployments.addPullSecretsOfSA(deployment);
+    imagePullSecrets.add(pullSecretOfSA);
+    assert deployment
```

Comment:
please don't use java `assert`. Use testng assert methods like in other tests

Reply:
assertNotNull for example